packages/caliper-core/test/worker/caliper-worker.js

```diff
@@ -161,5 +161,84 @@ describe('Caliper worker', () => {
             await worker.executeRound(mockTestMessage);
             validateCallsAndWarnings(4);
         });
+
+        it('should wait for all transactions to finish', async () => {
+            const mockRoundStats = {
+                getTotalFinishedTx: sinon.stub(),
+                getTotalSubmittedTx: sinon.stub()
+            };
+
+            mockRoundStats.getTotalFinishedTx.onCall(0).returns(0);
+            mockRoundStats.getTotalFinishedTx.onCall(1).returns(1);
+            mockRoundStats.getTotalSubmittedTx.returns(1);
+
+            await CaliperWorker._waitForTxsToFinish(mockRoundStats);
```

Comment:
We really want to avoid writing unit tests for private methods (ie methods that have the _ in fromt of them) we look to drive the testing at a public method level only. The only place in caliper where it looks like this is being broken is on the _sendSingleRequest method. However this is actually a method that has to be implemented by a connector and is called by something external so is infact a public method just not labelled as such.

Reply:
This is really useful, thanks for the insight